### PR TITLE
Fixed multiple errors that would render the project unusable

### PR DESCRIPTION
**Pull request by kri**
In a previous commit (see PR #12 and issue #9), it had been erroneously assumed that `https` could be used as a drop-in replacement for `request`, which is not the case at all.

In fact, the APIs differ so much that this project completely broke once the PR had been merged. It seems as if nobody had been able to notice this, since it took a quarter of a year until the maintainer had bumped the version to publish it on npm.

In conclusion, 1.2.1 is completely broken (it just crashes if you attempt to download anything with it), and 1.2.0 is the last version that actually works.

To remedy this, this PR ports the `request`-based code to `https`. I've also noticed a bug where in an option to `https.get`, the `headers` object had wrongly been named `header`, which prevented `https` from setting the correct headers.

### CODE REVIEW

**Comment by MrMaxie** [comment]
Ok, we bring back `request` because some cases (like following redirects, local requests etc) had some problems in HTTP and HTTPS in local tests etc.

Until 2.0 version (which will be written in TS and will have automated tests), just let's stay with `request`.

PS Sorry for the late response :(

**Comment by kri** [comment]
@MrMaxie Thanks for getting back to me.

Just in case you still want `request` gone, have a look at [follow-redirects](https://github.com/follow-redirects/follow-redirects), which exposes NodeJS's `http` or `https`, but follows redirects.

That way, you can keep the code from this PR, get redirects to work and get rid of the unnecessarily large weight from `request`.

PS: Don't worry about it. :)
